src/service/performance-impl.js

```diff
@@ -88,19 +88,8 @@ export class Performance {
     /** @private {string} */
     this.ampexp_ = '';
 
-    this.fcpDeferred_ = new Deferred();
-    this.fvrDeferred_ = new Deferred();
-    this.mbvDeferred_ = new Deferred();
-
-    // Platform service must be installed before performance serivce is
-    this.platform_ = Services.platformFor(this.win);
-
-    // TODO (micajuineho) change this once all platforms
-    // support PerformancePaintTiming
-    // https://developer.mozilla.org/en-US/docs/Web/API/PerformancePaintTiming
-    if (!this.platform_.isChrome() && !this.platform_.isOpera()) {
-      this.fcpDeferred_.resolve(null);
-    }
+    /** @private {Object} */
+    this.metrics_ = {};
```

Comment:
This seems very similar to the `signals` api we provide in `src/utils/signals.js`. Reusing that could save us some bytes.

Reply:
Thanks, I will look into the Signals API